webpack/assets/javascripts/react_app/components/common/TypeAheadSelect/__tests__/TypeAheadSelect.test.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import { shallow } from '@theforeman/test';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import TypeAheadSelect from '../';
+import { props, initialState } from '../TypeAheadSelect.fixtures';
+
+const mockStore = configureMockStore([thunk]);
+
+describe('TypeAheadSelect', () => {
+  it('renders', () => {
+    const wrapper = shallow(
+      <Provider store={mockStore(initialState)}>
+        <TypeAheadSelect {...props} />
+      </Provider>
+    );
+    expect(wrapper).toMatchSnapshot();
```

Comment:
This test file goes against Foreman's best practices.

This file (`TypeAheadSelect.test.js`) should import the dummy-component and contain the **unit-testing**, it should use the `testComponentSnapshotsWithFixtures`:
https://github.com/theforeman/foreman-js/tree/master/packages/test#unit-testing
https://sharvit.github.io/react-redux-test-utils/manual/unit-testing-components.html

Also, please create another file (`integration.test.js`) for the **integration-test**, see:
https://github.com/theforeman/foreman-js/tree/master/packages/test#integration-testing
https://sharvit.github.io/react-redux-test-utils/manual/integration-testing.html

Reply:
Added the integration tests but using testComponentSnapshotsWithFixtures here poses problems. The test run complains about having no provider for useDispatch and seems like I'll need to stub/mock all of the redux behaviors. That seems like a lot of hoops to jump through especially considering there are integration tests now. Any tips?

Reply:
Yes, that the reason we write unit-testing to the inner component that should be "dummy"

Reply:
@sharvit Since @jturel didn't modify the Patternfly component at all, I'm not even sure we need a unit test of the "dummy" component?  I trust Patternfly to test their components as needed... I'm thinking for our application, the integration test is enough.

Reply:
I'd agree with @jeremylenz on this one :)

Reply:
Shall I leave it as-is then? It's still nice to have a snapshot of the rendered component, I think

Reply:
Oh yes, I didn't notice it uses the pf-component.
Let's just remove this test file :+1:

Reply:
Don't you think there's still value in storing a snapshot of the rendered component? If the patternfly component changes we probably want to know about it even if it just means updating the snapshot

Reply:
@jturel If we do that then we should snapshot-test all the PF components, which I think would take too much time and effort.  Also, given the `<ForwardRef>` fiasco a few weeks ago, we definitely don't need to have those in order for Patternfly to break our snapshots :laughing:

Reply:
.. and it's actually only rendering the connected component anyway. removing!